apps/s_client.c

```diff
@@ -2841,10 +2916,23 @@ int s_client_main(int argc, char **argv)
         FD_ZERO(&readfds);
         FD_ZERO(&writefds);
 
-        if (SSL_is_dtls(con) && DTLSv1_get_timeout(con, &timeout))
-            timeoutp = &timeout;
-        else
+        if ((isdtls || isquic) && SSL_get_tick_timeout(con, &timeout)) {
+            /*
+             * TODO(QUIC): The need to do this seems like an unintended
```

Comment:
@hlandau - thoughts?

Reply:
I designed `SSL_get_tick_timeout` like this because `DTLSv1_get_tick_timeout` doesn't make a distinction between a failed call and an infinity value. In general I'd like to avoid adding new APIs which can't indicate a failure condition.

Reply:
@t-j-h what is your view on @hlandau's response?

Either way, if this requires further work, I think it should be handled by a different issue and not hold up this PR.

Reply:
Signalling there is no active timeout isn't an "error". A separate call can be used for that. Using negative value indicators and have variations across protocols is asking for confusion. Be more explicit in the handling. This is what leads to -2 and -3 return values getting added.

Reply:
Marking this as resolved as it is sorted in #20867.